### PR TITLE
check sessionToken in oauth2-proxy for kiabana sso 

### PR DESCRIPTION
**Pull request by amiya-elear**

### CODE REVIEW

File: oauthproxy.go [hunk 1]
```diff
@@ -858,9 +858,17 @@ func (p *OAuthProxy) Proxy(rw http.ResponseWriter, req *http.Request) {
...
+		// COCO: check session token present in cookie or not
+		cookie, err := req.Cookie("sessionToken")
+		if err != nil {
+			fmt.Printf("can't find sessionToken redireacting to logout page")
+			p.SignOut(rw, req)
+		}
 		// we are authenticated
 		p.addHeadersForProxying(rw, req, session)
 		p.serveMux.ServeHTTP(rw, req)
+		//<!--COCO End-->
```
Comment (agarwal13) [inline oauthproxy.go]: - this should be after the above if condition. i.e., after line # 867

File: oauthproxy.go [hunk 2]
```diff
@@ -858,9 +858,17 @@ func (p *OAuthProxy) Proxy(rw http.ResponseWriter, req *http.Request) {
 	session, err := p.getAuthenticatedSession(rw, req)
 	switch err {
 	case nil:
+		//<!--COCO Begin-->
+		// COCO: check session token present in cookie or not
+		cookie, err := req.Cookie("sessionToken")
+		if err != nil {
+			fmt.Printf("can't find sessionToken redireacting to logout page")
```
Comment (agarwal13) [inline oauthproxy.go]: - Why are we using fmt.Printf(), when there are no variables to be passed. Is this the default Print statement?

File: oauthproxy.go [hunk 3]
```diff
@@ -858,9 +858,17 @@ func (p *OAuthProxy) Proxy(rw http.ResponseWriter, req *http.Request) {
...
+			p.SignOut(rw, req)
```
Comment (agarwal13) [inline oauthproxy.go:866]: - There is no return after the p.SignOut?

**Review by agarwal13 — CHANGES_REQUESTED** [review]

File: oauthproxy.go (hunk 1 above)
Comment (amiya-elear) [reply oauthproxy.go]: handled

File: oauthproxy.go (hunk 2 above)
Comment (amiya-elear) [reply oauthproxy.go]: no, we can use logger.printf i think its better to use logger.Printf() many other places they use as logger message

File: oauthproxy.go (hunk 3 above)
Comment (amiya-elear) [reply oauthproxy.go:866]: yes its there .. but after logout, it's falling into the "/" redirect so I removed it.

**Review by agarwal13 — APPROVED** [review]

**Pull request merged**